src/Microsoft.Health.Fhir.CosmosDb/Configs/CosmosDataStoreParallelQueryOptions.cs

```diff
@@ -11,5 +11,10 @@ public class CosmosDataStoreParallelQueryOptions
         /// Gets the maximum degree of parallelism for the SDK to use when querying physical partitions in parallel.
         /// </summary>
         public int MaxQueryConcurrency { get; set; }
```

Comment:
After seeing this in the Cosmos code, thoughts on using a default based on the processor count? 

```suggestion
        public int MaxQueryConcurrency { get; set; } = Environment.ProcessorCount * 10;
```

/cc @Ivanidzo4ka also your thoughts?

Reply:
Sure, why not. TBH I'm not sure why fanning out queries happening on our side and not cosmos side, but if that's the case, why not.

Reply:
ok thanks, I am setting this to default. But just to mention we are overriding this with value 500 :)

Reply:
@Ivanidzo4ka this setting is the number of partitions they read from which == the number of Parallel tasks they create in the SDK :)